doc/feature/synapses.md

```diff
@@ -0,0 +1,189 @@
+    const float* preDistance() const;
+    const float* preSurfacePositionX() const;
+    const float* preSurfacePositionY() const;
+    const float* preSurfacePositionZ() const;
+    const float* preCenterPositionX() const;
+    const float* preCenterPositionY() const;
+    const float* preCenterPositionZ() const;
+
+    const uint32_t* postGID() const;
+    const uint32_t* postSectionID() const;
+    const uint32_t* postSegmentID() const;
+    const float* postDistance() const;
+    const float* postSurfacePositionX() const;
+    const float* postSurfacePositionY() const;
+    const float* postSurfacePositionZ() const;
+    const float* postCenterPositionX() const;
+    const float* postCenterPositionY() const;
+    const float* postCenterPositionZ() const;
+
+    const float* conductance() const;
+    const float* utilization() const;
+    const float* depression() const;
+    const float* facilitation() const;
+    const float* decay() const;
+    const int* efficacy() const;
+
+    // intersection
+    Synapses operator&( const Synapses& rhs ) const;
+
+    // union
+    Synapses operator+( const Synapses& rhs ) const;
+    Synapses operator|( const Synapses& rhs ) const;
+
+    // disjunction (XOR)
+    Synapses operator^( const Synapses& rhs ) const;
+
+    // difference
+    Synapses operator-( const Synapses& rhs ) const;
+};
+
+// sequentially & forwarding iteration through synapses for any number of GIDs
+class SynapsesStream
+{
+    bool eos() const;
+
+    // async load of synapses for count cells
+    // throws if current + count leads to eos
+    std::future< Synapses > read( size_t count = 1 ) const;
+};
+
+```
+
+## File format
+
+Uses existing nrn* files
+
+## Examples
+
+### Stream synapse positions for pathway density voxelization
+
+```cpp
+const brain::Circuit circuit;
+const brain::SynapsesStream& stream =
+   circuit.getSharedSynapses( circuit.getGIDs( "mtypetarget_1" ),
+                              circuit.getGIDs( "mtypetarget_2" ));
+std::future< brain::Synapses > future = stream.read();
+while( !stream.eos( ))
+{
+    const brain::Synapses synapses = future.get();
+    if( !stream.eos( ))
+        future = stream.read(); // fetch next
+
+    const float* __restrict__ posx = synapses.preSurfacePositionX();
+    const float* __restrict__ posy = synapses.preSurfacePositionY();
+    const float* __restrict__ posz = synapses.preSurfacePositionZ();
+    for( size_t i = 0; i < synapses.size(); ++i )
+        doVoxelization( posx[i], posy[i], posz[i] );
+}
+```
+
+### Access all afferent synapses of a GIDSet
+
+```cpp
+const brain::Circuit circuit;
+const brain::Synapses& synapses =
+    circuit.getAllAfferentSynapses( circuit.getGIDs( "Layer1" ));
+for( const auto& synapse : synapses )
+    std::cout << synapse.postGID() << " " << synapse.decay() << std::endl;
+```
+
+## Implementation
+
+* Data owned by container in the form of the brion::SynapseMatrix provided by
+  Brion, loaded upon construction
```

Comment:
While getting a first implementation for the synapse stream using position arrays, storing as matrices is not feasible. The data will be stored the same way like the BBPSDK Synapse_Dataset.